### ISSUE TITLE
Gamecube games (ISO's) can no longer be loaded from 2GB standard SD cards

### ISSUE THREAD

**Issue by paulness** [emukidid/swiss-gc#152]
SDGecko

SD cards checked
2GB Sandisk video (SDSDB-002G-B35)
2GB Transcend

All cards work for DOL files and homebrew, but not for Gamecube games (not even Animal Crossing). Zelda shows a strange green screen. Timesplitters 2 is frozen on the loading screen.

--
Older versions of Swiss can load both Gamecube games and homebrew DOL files from the standard Sandisk 2GB SD
Transcend is incompatible with older versions of Swiss altogether
32GB Sony SDHC works fine for loading Gamecube ISO's with the new Swiss

**Comment by NonaSuomy** [comment]
I can confirm this as well, SanDisk 2GB cards loaded up with 4 Swords backup, causes an actual kernel panic it seems, debug text dumps on the screen with a green boarder (EXCEPTION OCCURRED!) works fine on a 32GB card when I can get it to actually read without locking up on boot.

Others I've tried do the same as said in OP on 2GB cards.

I thought it might be a fat 16 thing but formatted the card fat32 and still same issue.

**Comment by ddribin** [comment]
I'm also unable to get Wind Waker to run from a 2GB SanDisk card.  I originally posted here:

  http://www.gc-forever.com/forums/viewtopic.php?f=25&t=3725

Here's my setup:

1. Swiss v0.4r382
2. Action Replay disc
3. SD card adapter I got from eBay (not SD Media Launcher)
4. SanDisk 2GB card, Type SD, Class 2, formatted FAT32

When I I try and run a Wind Waker ISO, Swiss says it loads 100%, but then it just goes to a black screen.

If I eject the 2GB and put in a SanDisk Ultra 32GB card with the same ISO, it works just fine. So it's something to do with the 2GB SD card.

**Comment by ghostkid2** [comment]
Out of curiosity exactly which '32GB Sony SDHC' card?
As I'd totally like to aquire the exact working model you speak of.

**Comment by paulness** [comment]
@ghostkid2 

Sony sf-32uy

**Comment by boktai1000** [comment]
I'm running into what I believe to be the same exact issue here and have tried a number of different variations, all to no avail with compatibility.

* Action Replay SD Card Adapter and Grey SD Gecko from eBay - Both work as they should for SWISS
* Two (2x) SanDisk for Wii 2GB SD Card and Two (2x) TOPRAM 4GB non-SHDC cards - No difference
* Various SWISS versions

Check out my comments on the forum post - https://www.gc-forever.com/forums/viewtopic.php?f=25&t=1375&p=39647#p39647

**Comment by boktai1000** [comment]
Hi @paulness did you ever find out what older SWISS versions worked for you? Very curious myself, and trying to provide information / feedback to @emukidid on a known working version.

Thanks!

**Comment by ChristinMilloy** [comment]
Confirming I am experiencing same issue. No games will run from the 2gb SD card that I boot swiss from on r431, tried seven very common games. The games will run fine once I swap a replacement SDHC card, all games run properly from the same image files.

**Comment by loriwood01** [comment]
Did this get resolved?  I saw it was just closed.

**Comment by boktai1000** [comment]
@loriwood01 I don't think it is as I tried SWISS r451 a couple months back and was still experiencing the same issues. I am not sure why the issue was closed @paulness

**Comment by fentenIT** [comment]
> @loriwood01 I don't think it is as I tried SWISS r451 a couple months back and was still experiencing the same issues. I am not sure why the issue was closed @paulness

Same here. Do you know which sd card is working right now?

**Comment by boktai1000** [comment]
> 
> 
> > @loriwood01 I don't think it is as I tried SWISS r451 a couple months back and was still experiencing the same issues. I am not sure why the issue was closed @paulness
> 
> Same here. Do you know which sd card is working right now?

None as far as I'm aware. I do not know a working configuration that involves 2GB. I have previously inquired in a comment here on what older versions of SWISS worked but did not get a reply on that. I do think knowing what version worked could help with finding where the regression happened. The older builds are no longer available for public download officially.

**Comment by emukidid** [comment]
It likely happened when I moved to fatFS instead of libfat. I haven't had any such regressions and all my SD cards still work apart from one FAT12 256MB SD card but that's no big loss.

**Comment by fentenIT** [comment]
what sd cards work for you guys? i have: 
Medion 2GB SD - which works only with swiss_r174
SanDisk SDHC 4GB - doesn't work
SanDisk Ultra SDHC 16GB - doesn't work

**Comment by boktai1000** [comment]
I don't have access to any SWISS builds other than the ones posted on GitHub, which only go back so far.

But I have four of the "2GB Wii SanDisk SD Cards" - I have tried SWISS versions ranging from r369 all the way up to r451 and haven't had any luck.

I also have two "4GB Non-SDHC TOPRAM SD Cards" that I purchased from eBay, and they also exhibit the same issues as the 2GB Wii SD cards.

To clarify, none of these cards are SDHC. I have also tried formatting them multiple ways, using the Panasonic SD Format Tool, using Windows, etc - all to no luck.

Further - SWISS does boot on these cards, and I can load homebrew on them fine, including emulators - it's the ISO loading that is specifically problematic.

**Comment by emukidid** [comment]
> I don't have access to any SWISS builds other than the ones posted on GitHub, which only go back so far.
> 
> But I have four of the "2GB Wii SanDisk SD Cards" - I have tried SWISS versions ranging from r369 all the way up to r451 and haven't had any luck.
> 
> I also have two "4GB Non-SDHC TOPRAM SD Cards" that I purchased from eBay, and they also exhibit the same issues as the 2GB Wii SD cards.
> 
> To clarify, none of these cards are SDHC. I have also tried formatting them multiple ways, using the Panasonic SD Format Tool, using Windows, etc - all to no luck.
> 
> Further - SWISS does boot on these cards, and I can load homebrew on them fine, including emulators - it's the ISO loading that is specifically problematic.

Now you're making sense, if you can browse it then the problem could be related to the in game code. Are the games on the smaller cards fragmented at all?

**Comment by boktai1000** [comment]
@emukidid just to clarify something before I test and respond, how are you checking for fragments and what are you running or should be done to prevent them? Are you referring to the built-in Windows Disk Defragment tool, or is there a separate tool that should be used with the SD Cards to analyze?

I can check and report back- I just want to make sure I'm understanding the terminology properly because it has been quite a many years since fragmentation was something that I had to worry about.

I do know that previously, I had tested with only putting a single ISO on the cards and not having any luck, but if fragmentation is still possible with a bare-bones SWISS install + ISO, then it's possible (but maybe unlikely?). Either way, let me know if you could and I'll come back with my findings.

**Comment by emukidid** [comment]
Confirmed to be broken, SD card block size is defaulting to 64 bytes just before Swiss goes in-game which is breaking the read code. I'll fix this shortly, I'd got it working but then got carried away in trying to optimise everything else around it.

**Comment by emukidid** [comment]
Try the latest release please: https://github.com/emukidid/swiss-gc/releases/latest

**Comment by ddribin** [comment]
I can confirm this now works in r533. Thank you!

**Comment by boktai1000** [comment]
Apologies on the late reply, but I had an opportunity to test this with Super Smash Bros Melee and can confirm I was able to load it successfully. I didn't do any sort of extensive testing, but it did indeed load, and I ran into some slowdown in the intro movie and loading, as well as a couple of minor audio pops but I don't believe those to be related to this issue. Anyways, again apologies on the late reply, but I feel as someone who was posting frequently about the issues in question here I should at least follow up on my end and wanted to say thank you as well for getting this fixed!
